tests/integrations/test_celery.py

```diff
@@ -0,0 +1,192 @@
+from contextlib import contextmanager
+import pytest
+import celery
+from celery import shared_task
+from celery.signals import task_failure
+from bugsnag.celery import connect_failure_handler, failure_handler
+from tests.utils import MissingRequestError
+
+
+@pytest.fixture(scope='function')
+def celery_config():
+    return {
+        'broker_url': 'memory://',
+        'result_backend': 'rpc',
+    }
+
+
+@contextmanager
+def celery_failure_handler():
+    """
+    The bugsnag celery integration works by listening to the celery
+    task_failure signal and sending an event when the signal is received.
+
+    This context manages the signal connection and ensures that error handling
+    does not occur across separate tests.
+    """
+    connect_failure_handler()
+    try:
+        yield
+        task_failure.disconnect(failure_handler)
+    except Exception:
+        task_failure.disconnect(failure_handler)
+        raise
```

Comment:
Could this be simplified? I think this is equivalent

```suggestion
    try:
        yield
    finally:
        task_failure.disconnect(failure_handler)
```

Reply:
it is! checked by breaking some stuff. 😄